src/main/java/net/jodah/failsafe/AbstractExecution.java

```diff
@@ -107,6 +107,12 @@ boolean complete(Object result, Throwable failure, boolean checkArgs) {
     }
 
     // Initialize or adjust the delay for backoffs
+    RetryPolicy.DelayFunction delayFunction = retryPolicy.getDelayFunction();
+    if (delayFunction != null) {
+        Duration dynamicDelay = delayFunction.calculateDelay(result, failure);
+        if (dynamicDelay != null && dynamicDelay.toNanos() >= 0)
+            delayNanos = dynamicDelay.toNanos();
+    }
     if (delayNanos == -1)
```

Comment:
If a `delayFunction` is present, do we want to still adjust for backoff? I'm thinking no - we just let the `delayFunction` do all of the work, in which case I think this could be an `else if`.